### PR TITLE
feat: add node roles to node list output vec-433

### PR DESCRIPTION
**Pull request by dwelch-spike**
Example

```
 asvec % ./asvec node ls
╭──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────╮
│                                                                 Nodes                                                                │
├───┬──────┬────────────────────────────┬────────────────┬─────────────────────┬─────────┬─────────────────────────────────────────────┤
│   │ NODE │ ROLES                      │ ENDPOINT       │          CLUSTER ID │ VERSION │ VISIBLE NODES                               │
├───┼──────┼────────────────────────────┼────────────────┼─────────────────────┼─────────┼─────────────────────────────────────────────┤
│ 1 │ LB   │ [INDEX_QUERY INDEX_UPDATE] │ 127.0.0.1:5000 │ 9205704725054754293 │ 1.0.0   │ {                                           │
│   │      │                            │                │                     │         │     18446651796337398664: [172.19.0.3:5000] │
│   │      │                            │                │                     │         │ }                                           │
╰───┴──────┴────────────────────────────┴────────────────┴─────────────────────┴─────────┴─────────────────────────────────────────────╯
```

### CODE REVIEW

File: e2e_multi_node_test.go [hunk 1]
```diff
@@ -55,16 +55,16 @@ func (suite *MultiNodeCmdTestSuite) TestNodeListCmd() {
 			"node ls with multiple nodes and seeds",
 			fmt.Sprintf("node ls --format 1 --no-color --seeds %s", suite.AvsHostPort.String()),
 			`Nodes
-,Node,Endpoint,Cluster ID,Version,Visible Nodes
-1,139637976803088,127.0.0.1:10000,<cluster-id>,<version>,"{
+,Node,Roles,Endpoint,Cluster ID,Version,Visible Nodes
```
Comment (arrowplum) [inline e2e_multi_node_test.go:58]: I think none of the tests have nodes with different roles do they? Just all of them default?

Comment (dwelch-spike) [reply e2e_multi_node_test.go:58]: The config for all the nodes is default so I think they all have the same roles.

File: cmd/writers/utils_test.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+		},
+		{
+			name: "TestRoleSecondary",
+			args: args{role: protos.NodeRole_INDEX_UPDATE},
+			want: "INDEX_UPDATE",
+		},
+		{
+			name: "TestRoleUnknown",
+			args: args{role: protos.NodeRole_KV_READ},
+			want: "KV_READ",
```
Comment (arrowplum) [inline cmd/writers/utils_test.go:32]: Do we ever get this one? This is more of a "Shouldn't happen" type of test right?

File: e2e_multi_node_test.go (hunk 1 above)
Comment (arrowplum) [reply e2e_multi_node_test.go:58]: I think you added tests for mixed nodes now right?

**Review by arrowplum — APPROVED** [review]
Looks good. I do have comment and question. (if I am understanding correctly)

File: cmd/writers/utils_test.go (hunk 2 above)
Comment (dwelch-spike) [reply cmd/writers/utils_test.go:32]: When I wrote this I though KV_READ was a valid return value. It is defined in the protobuf, so theoretically we could get it, but Ashish has said that it isn't exposed.

**Pull request merged**